keps/sig-network/4962-network-topology-standard/README.md

```diff
@@ -0,0 +1,867 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This document proposes a standard for declaring switch network topology in Kubernetes clusters,
+representing the hierarchy of nodes, switches, and interconnects.
+In this context, a `switch` can refer to a physical network device or a collection of such devices
+with close proximity and functionality.
+
+## Motivation
+
+With the rise of multi-node Kubernetes workloads that demand intensive inter-node communication,
+scheduling pods in close network proximity is becoming essential.
+Examples of such workloads include AI/ML training jobs or sets of interdependent, data-intensive services.
+
+However, Kubernetes currently lacks a standard method to describe cluster network topology, which is a key area for improvement.
+By establishing a consistent way to represent cluster network topology, this proposal lays the groundwork for
+advanced scheduling capabilities that take backend switched network topology and performance into account.
+
+Some major CSPs already offer mechanisms to discover instance network topology:
+- **Amazon AWS** provides the [DescribeInstanceTopology API](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeInstanceTopology.html).
+- **Google Cloud Platform (GCP)** exposes tools via the [Google Cloud SDK](https://cloud.google.com/go/docs/reference/cloud.google.com/go/compute/latest/apiv1), allowing the retrieval of rack and cluster IDs, which can be used to reconstruct network hierarchies.
+- **Oracle Cloud Infrastructure (OCI)** offers the [Capacity Topology API](https://docs.oracle.com/en-us/iaas/tools/oci-cli/3.50.1/oci_cli_docs/cmdref/compute/capacity-topology.html), providing topology-related information for their compute nodes.
+
+Beyond CSPs, certain on-premises clusters support network topology discovery, though this capability depends on the features of the underlying switch network vendors.
+
+An open-source project, [Topograph](https://github.com/NVIDIA/topograph), has implemented these approaches and is successfully deployed in production environments.
```

Comment:
It would be good to discuss what this proposal does differently than topograph?

- Is it porting over the general concepts? What worked/did not?
- Does it address the problems with the topograph project?

Reply:
This proposal establishes a standard for node labels.
Topograph will assign these labels to cluster nodes.